website/docs/reference/global-configs.md

```diff
@@ -103,6 +103,34 @@ config:
 
 </File>
 
+### Cache database objects for selected resource
+
+:::caution Experimental config flag
+This should be considered an _unstable_ config flag because it is experimental and subject to change. We reserve the right to make breaking changes to this config flag.
+:::
+
+When true, caches database objects relevant to selected resource. Otherwise, will cache all database objects related to the project.
```

Comment:
@jtcohen6 and possibly @gwenwindflower or @dbeatty10, can you add any context or color here? I'm also curious how or where you'd select this resource to be cached? Thanks!

Reply:
definitely!

dbt needs a bunch of metadata about all the actual database objects (relations in dbt) when you run commands, which requires hitting the database, which is always significantly slower than staying local, so it caches all this and relies on that cache when its able to for speed improvements (the specifics of that metadata and the conditions when it can use the cache vs needing to refresh it are not something i've given any thought to so i can't really go deeper than that -- jeremy and doug will be much more useful there).

if you're consistently working on a subset of your DAG though, you don't really need it to cache the whole project, just the part you're working on. this is not a big help on a small project, but could be a huge advantage on massive projects where a developer would never actually run the full project. 

so the selection here as to what is cached is tied to what the developer is selecting in their command(s) -- if i run `dbt build --select source:salesforce` to start working with all my salesforce models, only those will be cached, rather than the entire project as before -- and thus speed 🏃🏻‍♀️! performance 🏋🏻‍♀️!